### PR TITLE
Update project to require C++17

### PR DESCRIPTION
**Pull request by OsamaRab3**
enforce -std=c++17 support using AX_CXX_COMPILE_STDCXX_17

### CODE REVIEW

**Comment by OsamaRab3** [comment]
@antoninbas please review my PR

File: targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4 [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# =============================================================================
```
Comment (antoninbas) [inline targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4:1]: I think you also need https://www.gnu.org/software/autoconf-archive/ax_cxx_compile_stdcxx.html in this m4 directory?

File: m4/ax_cxx_compile_stdcxx.m4 [hunk 2]
```diff
@@ -0,0 +1,1018 @@
...
+
+// As C++20 supports feature test macros in the standard, there is no
+// immediate need to actually test for feature availability on the
+// Autoconf side.
+
+}  // namespace cxx20
+
+#endif  // __cplusplus < 202002L && !defined _MSC_VER
+
+]])
```
Comment (antoninbas) [inline m4/ax_cxx_compile_stdcxx.m4:1018]: file is missing a trailing newline character (applies to all the other new files as well)

**Review by antoninbas — COMMENTED** [review]
I wonder if the reference to C++11 in the README should be updated?

I see multiple CI failures for your PR.

**Comment by OsamaRab3** [comment]
> I see multiple CI failures for your PR.

I'm not sure about the exact cause of the CI failures at the moment. Could you please provide more details about the specific failures? I'd be happy to investigate further and work on resolving these issues. Any additional information or logs you can share as it will help me address the problems more effectively

**Comment by jafingerhut** [comment]
> > I see multiple CI failures for your PR.
> 
> I'm not sure about the exact cause of the CI failures at the moment. Could you please provide more details about the specific failures? I'd be happy to investigate further and work on resolving these issues. Any additional information or logs you can share as it will help me address the problems more effectively

I am pretty sure that the detailed log output of all CI tests, whether they are failing or successful, are publicly available to anyone.

If you look at this PR on github.com at this link https://github.com/p4lang/behavioral-model/pull/1266 and scroll down to the bottom, near the bottom you should see a list of tests with names like these, and a "Details" link to the right of each one:

+ Test / test (clang++-10, clang-10) (pull_request).  Failing after 12m
+ Test / test-ubuntu22 (pull_request) Failing after 1m

as well as some tests that have been successful.

**Comment by OsamaRab3** [comment]
@antoninbas Are there any additional changes or updates that need?

**Comment by antoninbas** [comment]
@OsamaRab3 At least one commit is not signed, so the DCO check is failing. I'd suggest squashing all 4 commits, and signing the resulting commit.

**Comment by OsamaRab3** [comment]
@antoninbas I reviewed the DCO check failure details and noticed that it was due to missing "Signed-off-by" lines in the commit messages. Following the instructions provided in the error message, I have rebased the  branch and added the necessary "Signed-off-by" lines to each commit.
Please let me know if there are any additional steps I should take or if further adjustments are needed

File: m4/ax_cxx_compile_stdcxx_17.m4 [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#
+#   Copying and distribution of this file, with or without modification, are
+#   permitted in any medium without royalty provided the copyright notice
+#   and this notice are preserved. This file is offered as-is, without any
+#   warranty.
+
+#serial 2
+
+AX_REQUIRE_DEFINED([AX_CXX_COMPILE_STDCXX])
+AC_DEFUN([AX_CXX_COMPILE_STDCXX_17], [AX_CXX_COMPILE_STDCXX([17], [$1], [$2])])
```
Comment (antoninbas) [inline m4/ax_cxx_compile_stdcxx_17.m4:35]: missing newline character

File: targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4 [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#
+#   Copying and distribution of this file, with or without modification, are
+#   permitted in any medium without royalty provided the copyright notice
+#   and this notice are preserved. This file is offered as-is, without any
+#   warranty.
+
+#serial 2
+
+AX_REQUIRE_DEFINED([AX_CXX_COMPILE_STDCXX])
+AC_DEFUN([AX_CXX_COMPILE_STDCXX_17], [AX_CXX_COMPILE_STDCXX([17], [$1], [$2])])
```
Comment (antoninbas) [inline targets/simple_switch_grpc/m4/ax_cxx_compile_stdcxx_17.m4:35]: ditto

**Comment by OsamaRab3** [comment]
@antoninbas any thing else?

**Review by antoninbas — APPROVED** [review]

**Pull request merged**